daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -1422,57 +1519,28 @@ private[lf] final class Compiler(
         case None => SEValue.None
         case Some(actors) => SEApp(SEBuiltin(SBSome), Array(actors))
       }
-      SEApp(
-        SEVal(ChoiceDefRef(tmplId, choiceId)),
-        Array(SEValue.bool(byKey), actors, contractId, argument))
+      SEApp(SEVal(ChoiceDefRef(tmplId, choiceId)), Array(actors, contractId, argument))
     }
 
   private[this] def compileExerciseByKey(
       tmplId: Identifier,
       key: SExpr,
       choiceId: ChoiceName,
-      // actors are either the singleton set of submitter of an exercise command,
-      // or the acting parties of an exercise node
-      // of a transaction under reconstruction for validation
+      // actors are only present when compiling old LF update expressions;
+      // they are computed from the controllers in newer versions
       optActors: Option[SExpr],
       argument: SExpr,
-  ): SExpr = {
-    // Translates 'exerciseByKey Foo <key> <choiceName> <optActors> <argument>' into:
-    // let key = <key>
-    // let maintainers = keyMaintainers key
-    // in \token ->
-    //    let coid = $fetchKey key maintainers token
-    //        exerciseResult = exercise coid coid <optActors> <argument> token
-    //    in exerciseResult
-    val template = lookupTemplate(tmplId)
+  ): SExpr =
+    // Translates 'A does exercise cid Choice with <params>'
```

Comment:
This doesn’t look quite right, this is still `exerciseByKey` but the description looks like `exercise`.

Reply:
fixed